primitives/forest-verifier/src/tests.rs

```diff
@@ -0,0 +1,1375 @@
+) {
+    let user_ids = vec![b"01"];
+    let bucket = b"bucket";
+    let file_name = b"sample64b";
+
+    let mut file_leaves = Vec::new();
+
+    println!("Chunking file into 32 byte chunks and building Merkle Patricia Tries...");
+
+    for user_id in user_ids {
+        let file_path = format!(
+            "{}-{}-{}.txt",
+            String::from_utf8(user_id.to_vec()).unwrap(),
+            String::from_utf8(bucket.to_vec()).unwrap(),
+            String::from_utf8(file_name.to_vec()).unwrap()
+        );
+
+        std::fs::create_dir_all(FILES_BASE_PATH).unwrap();
+        std::fs::File::create(FILES_BASE_PATH.to_owned() + &file_path).unwrap();
+
+        let file = std::fs::File::open(FILES_BASE_PATH.to_owned() + &file_path).unwrap();
+        let file_size = std::fs::File::metadata(&file).unwrap().len();
+        let (_memdb, fingerprint) = merklise_file::<LayoutV1<BlakeTwo256>>(&file_path);
+
+        let metadata = FileMetadata {
+            user_id,
+            bucket,
+            file_id: file_name,
+            size: file_size,
+            fingerprint: fingerprint
+                .as_ref()
+                .try_into()
+                .expect("slice with incorrect length"),
+        };
+
+        let metadata = bincode::serialize(&metadata).unwrap();
+        let metadata_hash = T::Hash::hash(&metadata);
+
+        file_leaves.push((metadata_hash, metadata));
+    }
+
+    // Construct the Merkle Patricia Forest
+    let mut memdb = MemoryDB::<T::Hash>::default();
+    let mut root: HashT<T> = Default::default();
+
+    let mut file_keys = Vec::new();
+    {
+        let mut merkle_forest_trie = TrieDBMutBuilder::<T>::new(&mut memdb, &mut root).build();
+
+        // Insert file leaf and metadata into the Merkle Patricia Forest.
+        for file in &file_leaves {
+            merkle_forest_trie
+                .insert(file.0.as_ref(), file.1.as_ref())
+                .unwrap();
+
+            file_keys.push(file.0);
+        }
+
+        println!(
+            "Merkle Patricia Forest Trie root: {:?}",
+            merkle_forest_trie.root()
+        );
+    }
+
+    // Sorting file keys for deterministic proof generation
+    file_keys.sort();
+
+    (memdb, root, file_keys)
+}
+
+/// Chunk a file into [`CHUNK_SIZE`] byte chunks and store them in a Merkle Patricia Trie.
+///
+/// The trie is stored in a [`MemoryDB`] and the [`Root`] is returned.
+///
+/// TODO: make this function fetch data from storage using Storage trait.
+/// _It is assumed that the file is located in the same directory as the executable._
+pub fn merklise_file<T: TrieLayout>(file_path: &str) -> (MemoryDB<T::Hash>, HashT<T>) {
+    let file_path = FILES_BASE_PATH.to_owned() + file_path;
+    let mut file = std::fs::File::open(file_path).unwrap();
+    let mut buf = [0; CHUNK_SIZE];
+    let mut chunks = Vec::new();
+
+    // create list of key-value pairs consisting of chunk metadata and chunk data
+    loop {
```

Comment:
This is used for testing but if you decide to move it to `FileDataTrie`,  it should not read everything in memory but construct it as you go. See [this](https://github.com/Moonsong-Labs/storage-hub/pull/72/files#diff-f5b58703c6605ce51568c53912a42386c530956f2f22e15c0a4bcaa4bdfbdf65R89).